src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -727,6 +736,27 @@ func (r *StatefulSetResource) getPorts() []corev1.ContainerPort {
 	return ports
 }
 
+func (r *StatefulSetResource) getStartupProbe() *corev1.Probe {
+	if r.pandaCluster.AdminAPIInternal() == nil {
+		return nil
+	}
+	return &corev1.Probe{
+		Handler: corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Host:   r.serviceFQDN,
+				Path:   "v1/status/ready",
+				Port:   intstr.FromInt(r.pandaCluster.AdminAPIInternal().Port),
```

Comment:
also I think this won't work on installations that have TLS on internal listener

Reply:
As agreed offline, the operator will only add a Startup Probe to clusters with Internal Admin API without TLS/mTLS enabled, which is the default configuration.